execution_engine/src/core/engine_state/mod.rs

```diff
@@ -460,19 +496,20 @@ where
     ///
     /// Currently a special shortcut is taken to distinguish a native transfer, from a deploy.
     ///
-    /// Return execution results which contains results from each deploy ran.
+    /// Returns an `Execution` which contains results from each deploy executed.
     pub fn run_execute(
         &self,
         correlation_id: CorrelationId,
         mut exec_request: ExecuteRequest,
-    ) -> Result<ExecutionResults, Error> {
+    ) -> Result<Execution, Error> {
         let executor = Executor::new(*self.config());
 
-        let deploys = exec_request.take_deploys();
-        let mut results = ExecutionResults::with_capacity(deploys.len());
+        let deploy_items = exec_request.take_deploys();
+        let mut exec_results = BTreeMap::new();
```

Comment:
**[blocker]:** This needs to be a `Vec` to retain the order of transforms. Otherwise, we might `Write` a transform that's not issued by the last transaction in a block 😬

Reply:
Done.